prime-router/src/main/kotlin/azure/ReportFunction.kt

```diff
@@ -290,18 +281,19 @@ class ReportFunction : Logging {
             ?: request.queryParameters[PAYLOAD_NAME_PARAMETER]
     }
 
-    // TODO: Make this so that we check sender's configuration if there is no param type, this is blocked by
-    //  adding the 'processingType' attribute to a sender
-    private fun processingType(request: HttpRequestMessage<String?>): ProcessingType {
+    private fun processingType(request: HttpRequestMessage<String?>, sender: Sender): ProcessingType {
         // uppercase first char, so it matches enum when 'async' is passed in
-        val processingTypeString = request.queryParameters.getOrDefault(PROCESSING_TYPE_PARAMETER, "Sync")
-            .replaceFirstChar { it.uppercase() }
-        val processingType = try {
-            ProcessingType.valueOf(processingTypeString)
-        } catch (e: IllegalArgumentException) {
-            ProcessingType.Sync
+        val processingTypeString = request.queryParameters.get(PROCESSING_TYPE_PARAMETER)
+            ?.replaceFirstChar { it.uppercase() }
+        return if (processingTypeString == null) {
+            sender.processingType
+        } else {
+            try {
+                ProcessingType.valueOf(processingTypeString)
+            } catch (e: IllegalArgumentException) {
+                sender.processingType
```

Comment:
I feel like if the sender submits a badly formed value, we should flag that as an error and do nothing, rather than just quietly override.  Otherwise the erroneous value never gets fixed.

Reply:
I woudl say that 'sync', 'async', 'Sync', 'Async', 'SYNC', 'ASYNC' or any combination are all valid values...it is just a flag so I don't think we really care the capitalization as long as it is those letters in that order

Reply:
I'm going to tack on returning the error to #3431, the refactoring there will make that trivial.